packages/block-library/src/template-part/edit/selection/template-part-previews.js

```diff
@@ -111,23 +111,37 @@ function TemplatePartsByArea( {
 	area = 'uncategorized',
 	labelsByArea,
 } ) {
+	const templatePartsToShow =
+		templateParts.filter(
+			( templatePart ) =>
+				'uncategorized' === area || templatePart.area === area
+		) || [];
 	const templatePartsByArea = useMemo( () => {
-		return Object.values( groupBy( templateParts, 'area' ) );
+		return Object.values( groupBy( templatePartsToShow, 'area' ) );
 	}, [ templateParts, area ] );
+
 	const currentShownTPs = useAsyncList( templateParts );
 
+	if ( ! templatePartsToShow.length ) {
+		return (
+			<PanelGroup
+				title={ `${ __( 'Area' ) }: ${ labelsByArea[ area ] }` }
```

Comment:
I'm not an i18n expert, but I think it would be more flexible if we used a string token here and incorporate the colon into the translatable string. Easier to read as well.
```suggestion
				title={ sprintf( __( 'Area: %s'), labelsByArea[ area ] ) }
```

Reply:
I think that makes a lot more sense. thanks!